### PR TITLE
Convert to TF 0.12. Add tests. Add Codefresh test pipeline. 

### PR DESCRIPTION
**Pull request by maximmi**
## what
* Port module to Terraform 0.12
* Pin all providers
* Add example for testing
* Add `bats` and `terratest` for the example
* Add Codefresh badge to point to the test pipeline in `terraform-modules` project
* Update README 

## why
* Module currently does not work with 0.12. Much easier syntax
* Better regression control
* Automatically test the example on every commit and pull request
* Provision resources on AWS in the test account and check the outputs for the correct values
* `terraform-modules` project contains pipelines for all terraform modules

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+## what
```
Comment (johncblandii) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: #nit markdown lint.

```suggestion
## what

```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+* Describe high-level what changed as a result of these commits (i.e. in plain-english, what do these changes mean?)
+* Use bullet points to be concise and to the point.
+
+## why
```
Comment (johncblandii) [inline .github/PULL_REQUEST_TEMPLATE.md:5]: #nit markdown lint.

```suggestion
## why

```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+* Provide the justifications for the changes (e.g. business case). 
+* Describe why these changes were made (e.g. why do these commits fix the problem?)
+* Use bullet points to be concise and to the point.
+
+## references
```
Comment (johncblandii) [inline .github/PULL_REQUEST_TEMPLATE.md:10]: #nit markdown lint.

```suggestion
## references

```

File: variables.tf [hunk 4]
```diff
@@ -53,48 +53,49 @@ variable "customer_gateway_bgp_asn" {
...
+  type        = list(string)
   description = "List of CIDR blocks to be used as destination for static routes. Routes to destinations will be propagated to the route tables defined in `route_table_ids`"
   default     = []
 }
 
 variable "vpn_connection_tunnel1_inside_cidr" {
-  type        = "string"
+  type        = string
   description = "The CIDR block of the inside IP addresses for the first VPN tunnel"
   default     = ""
```
Comment (johncblandii) [inline variables.tf:81]: I found these throw errors if they're empty so I had to set them to `null`. Consumers shouldn't need to do that.

```suggestion
  default     = null
```

File: variables.tf [hunk 5]
```diff
@@ -53,48 +53,49 @@ variable "customer_gateway_bgp_asn" {
...
 }
 
 variable "vpn_connection_tunnel2_inside_cidr" {
-  type        = "string"
+  type        = string
   description = "The CIDR block of the inside IP addresses for the second VPN tunnel"
   default     = ""
```
Comment (johncblandii) [inline variables.tf:87]:
```suggestion
  default     = null
```

File: variables.tf [hunk 6]
```diff
@@ -53,48 +53,49 @@ variable "customer_gateway_bgp_asn" {
...
 }
 
 variable "vpn_connection_tunnel1_preshared_key" {
-  type        = "string"
+  type        = string
   description = "The preshared key of the first VPN tunnel. The preshared key must be between 8 and 64 characters in length and cannot start with zero. Allowed characters are alphanumeric characters, periods(.) and underscores(_)"
   default     = ""
```
Comment (johncblandii) [inline variables.tf:93]:
```suggestion
  default     = null
```

File: variables.tf [hunk 7]
```diff
@@ -53,48 +53,49 @@ variable "customer_gateway_bgp_asn" {
...
 }
 
 variable "vpn_connection_tunnel2_preshared_key" {
-  type        = "string"
+  type        = string
   description = "The preshared key of the second VPN tunnel. The preshared key must be between 8 and 64 characters in length and cannot start with zero. Allowed characters are alphanumeric characters, periods(.) and underscores(_)"
   default     = ""
```
Comment (johncblandii) [inline variables.tf:99]:
```suggestion
  default     = null
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (maximmi) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: we have to change it here:

File: variables.tf (hunk 5 above)
Comment (maximmi) [reply variables.tf]: done

**Pull request merged**